tinker_cookbook/tests/test_renderers.py

```diff
@@ -935,3 +956,565 @@ def test_deepseek_post_tool_formatting():
+    content = "<think>step 1</think>partial<think>step 2</think>final"
+    parts = parse_content_blocks(content)
+
+    assert len(parts) == 4
+    assert parts[0] == ThinkingPart(type="thinking", thinking="step 1")
+    assert parts[1] == TextPart(type="text", text="partial")
+    assert parts[2] == ThinkingPart(type="thinking", thinking="step 2")
+    assert parts[3] == TextPart(type="text", text="final")
+
+
+def test_parse_content_blocks_empty_blocks_omitted():
+    """Test parse_content_blocks omits empty think blocks."""
+    parts = parse_content_blocks("<think></think>visible")
+
+    assert len(parts) == 1
+    assert parts[0]["type"] == "text"
+    assert parts[0]["text"] == "visible"
+
+
+def test_parse_content_blocks_whitespace_handling():
+    """Test parse_content_blocks handles whitespace correctly."""
+    parts = parse_content_blocks("<think>  thinking  </think>  answer  ")
+
+    assert len(parts) == 2
+    # Thinking content preserves internal whitespace but text gets stripped
+    assert parts[0]["thinking"] == "  thinking  "
+    assert parts[1]["text"] == "answer"
+
+
+def test_parse_content_blocks_tool_call_only():
+    """Test parse_content_blocks parses tool calls."""
+    content = '<tool_call>{"name": "search", "arguments": {"query": "test"}}</tool_call>'
+    parts = parse_content_blocks(content)
+
+    assert len(parts) == 1
+    assert parts[0]["type"] == "tool_call"
+    tool_call = parts[0]["tool_call"]
+    assert tool_call.function.name == "search"
+    assert tool_call.function.arguments == '{"query": "test"}'
+
+
+def test_parse_content_blocks_interleaved():
+    """Test parse_content_blocks handles interleaved think and tool_call blocks."""
+    content = '<think>Let me search</think>Searching...<tool_call>{"name": "search", "arguments": {"q": "test"}}</tool_call>Done'
+    parts = parse_content_blocks(content)
+
+    assert len(parts) == 4
+    assert parts[0] == ThinkingPart(type="thinking", thinking="Let me search")
+    assert parts[1] == TextPart(type="text", text="Searching...")
+    assert parts[2]["type"] == "tool_call"
+    assert parts[2]["tool_call"].function.name == "search"
+    assert parts[3] == TextPart(type="text", text="Done")
+
+
+def test_parse_content_blocks_invalid_tool_call():
+    """Test parse_content_blocks handles invalid tool call JSON as UnparsedToolCallPart."""
+    content = '<tool_call>not valid json</tool_call>text after'
+    parts = parse_content_blocks(content)
+
+    # Invalid tool call is included as UnparsedToolCallPart, text is still captured
+    assert len(parts) == 2
+    assert parts[0]["type"] == "unparsed_tool_call"
+    assert "Invalid JSON" in parts[0]["error"]
+    assert parts[1] == TextPart(type="text", text="text after")
+
+
+# =============================================================================
+# Qwen3 parse_response Tests
+# =============================================================================
```

Comment:
Please shorten this code by removing trivial assertions and tests. Just keep tests that are checking some code that involves tricky logic.